src/main/java/htsjdk/samtools/reference/FastaReferenceWriter.java

```diff
@@ -0,0 +1,758 @@
+    }
+
+    /**
+     * Appends a new sequence to the output with or without a description and an alternative number of bases-per-line.
+     * <p>
+     * This is a convenient short handle for {@code startSequence(name, description, bpl).appendBases(bases)}.
+     * </p>
+     * <p>
+     * A {@code null} or empty ("") description will be ignored (no description will be output).
+     * </p>
+     * <p>
+     * The new sequence remains open meaning that additional bases for that sequence can be added with additional calls to {@link #appendBases}.
+     * </p>
+     *
+     * @param name         the name of the new sequence.
+     * @param bases        the (first) bases of the sequence.
+     * @param description  the description for the sequence.
+     * @param basesPerLine alternative number of bases per line to be used for the sequence.
+     * @return a reference to this very same writer.
+     * @throws IOException              if such an exception is thrown when actually writing into the output streams/channels.
+     * @throws IllegalArgumentException if either {@code name} or {@code bases} is {@code null} or contains an invalid value (e.g. unsupported bases or sequence names). Also when the
+     *                                  {@code description} contains unsupported characters or {@code basesPerLine} is 0 or negative.
+     * @throws IllegalStateException    if the writer is already closed, a previous sequence (if any was opened) has no base appended to it or a sequence
+     *                                  with such name was already appended to this writer.
+     */
+    public FastaReferenceWriter appendSequence(final String name, final String description, final int basesPerLine, final byte[] bases) throws IOException {
+        return startSequence(name, description, basesPerLine).appendBases(bases);
+    }
+
+    private void assertSequenceOpen() {
+        if (currentSequenceName == null) {
+            throw new IllegalStateException("trying to add bases without starting a sequence");
+        }
+    }
+
+    private void assertIsNotClosed() {
+        if (closed) {
+            throw new IllegalStateException("already closed");
+        }
+    }
+
+    /**
+     * Closes this writer flushing all remaining writing operation input the output resources.
+     * <p>
+     * Further calls to {@link #appendBases} or {@link #startSequence} will result in an exception.
+     * </p>
+     *
+     * @throws IOException if such exception is thrown when closing output writers and output streams.
+     */
+    public void close() throws IOException {
+        if (!closed) {
+            closeSequence();
+            if (sequenceNamesAndSizes.isEmpty()) {
+                throw new IllegalStateException("no sequences where added to the reference");
+            }
+            fastaStream.close();
+            indexWriter.close();
+            dictWriter.close();
+            closed = true;
```

Comment:
Maybe put this in a `finally` in case other `close()` calls above throw? Once we've tried to close ourselves and failed we don't want to try again.